02-intermediate-ruby/api-post.md

```diff
@@ -0,0 +1,282 @@
+
+Now hit `send`
+
+**Questions**
+
+- What JSON is returned by Slack?  What status code is returned?
+- Try submitting an invalid token, what happens?
+
+You've now successfully posted to Slack from Postman!
+
+#### Thinking About Testing
+
+**With a partner** discuss what are the things to test when sending messages to slack.  Consider the elements of the body section as arguments to a method, how can they vary.
+
+- What is the nominal case?
+- Write out all the edge-cases you can think of.  What could go wrong when sending messages to Slack?
+  - Try them out in Postman to see the results.
+
+### Using HTTParty - Setting Up
+
+In this lesson we will build a method for sending messages to Slack in a TDD fashion.  Start by forking and cloning the [Ada Gold slack-cli](https://github.com/AdaGold/slack-cli) repository.  
+
+### What Do We Have?
+
+In this repository we have a `Rakefile` to run our tests with, a `run_chatstream.rb` file to run our finished application, and a `specs` folder with the following files?
+
+- `spec_helper.rb`
+- `slack_api_wrapper_spec.rb`
+
+The `lib` directory also has a `slack_api_wrapper.rb` file with the following code inside:
+
+```ruby
+# slack_api_wrapper.rb
+require 'httparty'
+
+class SlackApiWrapper
+  BASE_URL = 'https://slack.com/api/'
+
+  # Code goes here
+
+end
+```
+
+In this exercise we will build some tests for a `send_msg` method in the `SlackApiWrapper` class and then build out the functionality to satisfy the tests.  
+
+### Our First Test - Nominal Case
+
+For our first test we will build a test case for sending a message to Slack successfully.  We will need to set up the testing first.  Add the following to `specs/spec_helper.rb`
+
+```ruby
+# spec_helper.rb
+require 'dotenv'
+Dotenv.load
+
+require 'minitest/autorun'
+require 'minitest/reporters'
+require 'vcr'
+require 'webmock/minitest'
+require_relative '../lib/slack_api_wrapper'
+
+Minitest::Reporters.use! Minitest::Reporters::SpecReporter.new
+
+VCR.configure do |config|
+  config.cassette_library_dir = 'specs/cassettes' # folder where casettes will be located
+  config.hook_into :webmock # tie into this other tool called webmock
+  config.default_cassette_options = {
+    record: :new_episodes,    # record new data when we don't have it yet
+    match_requests_on: [:method, :uri, :body] # The http method, URI and body of a request all need to match
+  }
+  # Don't leave our Slack token lying around in a cassette file.
+  config.filter_sensitive_data("SLACK_TOKEN") do
+    ENV["SLACK_TOKEN"]
+  end
+end
+```
+
+This code configures VCR just like we did when we tested `GET` requests.  We have added `config.filter_sensitive_data("SLACK_TOKEN")` so that anytime our VCR sees our slack token it will replace it with the words "SLACK_TOKEN".  **This will prevent the token from appearing in git**.  
+
+We can add our Slack token to a `.env` file in the project's root folder like this:
+
+```
+SLACK_TOKEN: <TOKEN_GOES_HERE>
```

Comment:
I'm fairly certain that the syntax we've used before uses an `=` instead of a `:`, any reason why the switch or is this a typo?